File: U1/EXAMEN_SOAP_DOTNET_GR01/01 SERVIDOR/Comercializadora_Soap_DotNet_GR01/Comercializadora_Soap_DotNet_GR01/DTOs/CrearProductoDTO.cs
using System.Runtime.Serialization;

namespace Comercializadora_Soap_DotNet_GR01.DTOs
{
    /// <summary>
    /// DTO para crear un producto (sin campos autogenerados)
    /// </summary>
    [DataContract]
    public class CrearProductoDTO
    {
        [DataMember(Order = 1)]
        public string Codigo { get; set; }

        [DataMember(Order = 2)]
        public string Nombre { get; set; }

        [DataMember(Order = 3)]
        public string Descripcion { get; set; }

        [DataMember(Order = 4)]
        public decimal Precio { get; set; }

        [DataMember(Order = 5)]
        public int Stock { get; set; }

        [DataMember(Order = 6)]
        public string Categoria { get; set; }

        [DataMember(Order = 7)]
        public string ImagenUrl { get; set; }
    }
}

File: U2/PROYECTO/04. SERVIDOR/Comercializadora_RestFul_Java_GR01/src/main/java/ec/edu/monster/DTOs/CrearProductoDTO.java
package ec.edu.monster.DTOs;

import lombok.Data;
import java.math.BigDecimal;

/**
 * DTO para crear un producto
 */
@Data
public class CrearProductoDTO {
    private String codigo;
    private String nombre;
    private String descripcion;
    private BigDecimal precio;
    private Integer stock;
    private String categoria;
    private String imagenUrl;
}
